base/applications/cmdutils/taskkill/taskkill.c

```diff
@@ -226,6 +294,11 @@ static int send_close_messages(void)
     {
         WCHAR *p = task_list[i];
         BOOL is_numeric = TRUE;
+        DWORD *pkill_list = HeapAlloc(GetProcessHeap(), 0, pid_list_size * sizeof(DWORD));
```

Comment:
It appears that `pid_list_size` is not modified while running this `for (i = 0; i < task_count; i++)` loop.
Therefore for each iteration of this loop, `pkill_list` is reallocated with exactly the same number of elements.
Thus I think you could save all that and allocate only once this array, just after having called the `enumerate_processes()` function, and then within this for-loop you just zero-out the contents of this list at each pass. You would then free the allocated memory for this list outside of the loop.

Reply:
Yes, I agree with you. I changed the code.